### PR TITLE
🗞️ Adding in OFTAdapter to create-lz-oapp

### PR DESCRIPTION
**Pull request by sirarthurmoney**
In this PR:

- Adding in the ability for create-lz-app to generate OFTAdapter

### CODE REVIEW

**Review by janjakubnanista — APPROVED** [review]

**Review by janjakubnanista — APPROVED** [review]

File: examples/oft-adapter/contracts/MyOFTAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFTAdapter } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTAdapter.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/contracts/MyOFTAdapter.sol:5]: Use the ones in `@layerzerolabs/oft-evm` instead.

**Review by TRileySchwarz — APPROVED** [review]

**Review by janjakubnanista — APPROVED** [review]

**Review by ryandgoulding — APPROVED** [review]

File: examples/oft-adapter/test/foundry/MyOFTAdapter.t.sol [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+// OApp imports
+import { IOAppOptionsType3, EnforcedOptionParam } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oapp/libs/OAppOptionsType3.sol";
+import { OptionsBuilder } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oapp/libs/OptionsBuilder.sol";
+
+// OFT imports
+import { IOFT, SendParam, OFTReceipt } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/interfaces/IOFT.sol";
+import { MessagingFee, MessagingReceipt } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTCore.sol";
+import { OFTMsgCodec } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/libs/OFTMsgCodec.sol";
+import { OFTComposeMsgCodec } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/libs/OFTComposeMsgCodec.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/test/foundry/MyOFTAdapter.t.sol:18]: these need to be changed

File: examples/oft-adapter/test/mocks/OFTAdapterMock.sol [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFTAdapter } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTAdapter.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/test/mocks/OFTAdapterMock.sol:5]: and this.

File: examples/oft-adapter/test/mocks/OFTMock.sol [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFT } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFT.sol";
+import { SendParam } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTCore.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/test/mocks/OFTMock.sol:6]: and these

File: examples/oft-adapter/contracts/MyOFT.sol [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFT } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFT.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/contracts/MyOFT.sol:5]: and this

File: examples/oft-adapter/contracts/MyOFTAdapter.sol [hunk 6]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFTAdapter } from "@layerzerolabs/oft-evm/contracts/OFTAdapter.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/contracts/MyOFTAdapter.sol:5]: and this.

File: examples/oft-adapter/contracts/mocks/MyOFTMock.sol [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFT } from "@layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFT.sol";
```
Comment (ryandgoulding) [inline examples/oft-adapter/contracts/mocks/MyOFTMock.sol:5]: and this.

**Review by ryandgoulding — CHANGES_REQUESTED** [review]

**Review by ryandgoulding — APPROVED** [review]

**Review by ryandgoulding — APPROVED** [review]

**Pull request merged**